storage/integration_test.go

```diff
@@ -744,6 +746,129 @@ func TestIntegration_ObjectsRangeReader(t *testing.T) {
 	}
 }
 
+func TestIntegration_ObjectReadGRPC(t *testing.T) {
+	// TODO: Remove this once v2 rollout is complete.
+	t.SkipNow()
+
+	ctx := context.Background()
+
+	// Create an HTTP client to upload test data.
+	hc := testConfig(ctx, t)
+	defer hc.Close()
+
+	content := []byte("Hello, world this is a grpc request")
+
+	// Upload test data.
+	name := uidSpace.New()
+	ho := hc.Bucket(grpcBucket).Object(name)
+	if err := writeObject(ctx, ho, "text/plain", content); err != nil {
+		t.Fatal(err)
+	}
+
+	// Initialize gRPC client for Read testing.
+	client, err := newClientWithGRPC(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	obj := client.Bucket(grpcBucket).Object(name)
+	defer func() {
+		if err := obj.Delete(ctx); err != nil {
+			log.Printf("failed to delete test object: %v", err)
+		}
+	}()
+
+	// Using a negative length to indicate reading to the end.
+	r, err := obj.NewRangeReader(ctx, 0, -1)
+	if err != nil {
+		t.Error(err)
+	}
+	defer r.Close()
+
+	b := new(bytes.Buffer)
+	b.Grow(len(content))
+
+	n, err := io.Copy(b, r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if n == 0 {
+		t.Fatal("Expected to have read more than 0 bytes")
+	}
+
+	got := b.String()
+	want := string(content)
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("got(-),want(+):\n%s", diff)
+	}
+}
+
+func TestIntegration_ObjectReadChunksGRPC(t *testing.T) {
+	// TODO: Remove this once v2 rollout is complete.
+	t.SkipNow()
+
+	ctx := context.Background()
+
+	// Create an HTTP client to upload test data.
+	hc := testConfig(ctx, t)
+	defer hc.Close()
+
+	// Use a larger blob to test chunking logic. This is a little over 5MB.
+	content := bytes.Repeat([]byte("a"), 5<<20)
+
+	// Upload test data.
+	name := uidSpace.New()
+	ho := hc.Bucket(grpcBucket).Object(name)
+	if err := writeObject(ctx, ho, "text/plain", content); err != nil {
+		t.Fatal(err)
+	}
+
+	// Initialize gRPC client for Read testing.
+	client, err := newClientWithGRPC(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	obj := client.Bucket(grpcBucket).Object(name)
+	defer func() {
+		if err := obj.Delete(ctx); err != nil {
+			log.Printf("failed to delete test object: %v", err)
+		}
+	}()
+
+	r, err := obj.NewReader(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r.Close()
+
+	bufSize := len(content)
+	buf := make([]byte, bufSize)
+
+	// Read in 4KB chunks.
+	chunk := 4 << 10
```

Comment:
You may want to read in sizes that are *not* a divisor of the received chunk size, just to make sure you can read across a `recv()` boundary.

Reply:
Good idea. Done.